DSharpPlus/Clients/DiscordClient.Dispatch.cs

```diff
@@ -1775,6 +1832,175 @@ internal async Task OnVoiceServerUpdateEventAsync(string endpoint, string token,
+                    CurrentMember = cthread.CurrentMember,
+                    //threadMembers,
+                };
+
+                updateEvent = new ThreadUpdateEventArgs
+                {
+                    ThreadAfter = thread,
+                    ThreadBefore = new Optional<DiscordThreadChannel>(threadOld),
+                    Guild = thread.Guild,
+                    Parent = thread.Parent
+                };
+            }
+            else
+            {
+                updateEvent = new ThreadUpdateEventArgs
+                {
+                    ThreadAfter = thread,
+                    Guild = thread.Guild,
+                    Parent = thread.Parent
+                };
+                guild._threads[thread.Id] = thread;
+            }
+
+            await this._threadUpdated.InvokeAsync(this, updateEvent).ConfigureAwait(false);
+        }
+
+        internal async Task OnThreadDeleteEventAsync(DiscordThreadChannel thread)
+        {
+            if (thread == null)
+                return;
+
+            thread.Discord = this;
+
+            var gld = thread.Guild;
+            if (gld._threads.TryRemove(thread.Id, out var cachedThread))
+                thread = cachedThread;
+
+            await this._threadDeleted.InvokeAsync(this, new ThreadDeleteEventArgs { Thread = thread, Guild = thread.Guild, Parent = thread.Parent}).ConfigureAwait(false);
+        }
+
+        internal async Task OnThreadListSyncEventAsync(DiscordGuild guild, IReadOnlyList<ulong?> channel_ids, IReadOnlyList<DiscordThreadChannel> threads, IReadOnlyList<DiscordThreadChannelMember> members)
+        {
+            guild.Discord = this;
+
+            var channels = channel_ids.Select(x => guild.GetChannel(x.Value)); //getting channel objects
+            foreach (var channel in channels)
+            {
+                channel.Discord = this;
+            }
+            foreach(var thread in threads)
+            {
+                thread.Discord = this;
+                guild._threads[thread.Id] = thread;
+            }
+            foreach(var member in members)
+            {
+                member.Discord = this;
+                member._guild_id = guild.Id;
+                var thread = threads.SingleOrDefault(x => x.Id == member.ThreadId);
+                if (thread != null)
+                    thread.CurrentMember = member;
+            }
+
+            await this._threadListSynced.InvokeAsync(this, new ThreadListSyncEventArgs { Guild = guild, Channels = channels.ToList().AsReadOnly(), Threads = threads, CurrentMembers = members.ToList().AsReadOnly() }).ConfigureAwait(false);
+        }
+
+        internal async Task OnThreadMemberUpdateEventAsync(DiscordThreadChannelMember member)
+        {
+            member.Discord = this;
+
+            var thread = this.InternalGetCachedThread(member.ThreadId);
+            member._guild_id = thread.Guild.Id;
+            thread.CurrentMember = member;
```

Comment:
Shouldn't this follow the [behavior defined in DiscordGuild.cs](https://github.com/DSharpPlus/DSharpPlus/blob/master/DSharpPlus/Entities/Guild/DiscordGuild.cs#L464)?

Reply:
Not to my understanding. Current member is a singular member representing whether or not the bot is in the thread. Its this way atm to stay more consistent with Discord's API.